R/kernel.r

```diff
@@ -129,6 +129,35 @@ handle_shell = function() {
         print(c('Got unhandled msg_type:', msg$header$msg_type)))
 },
 
+handle_abort = function(){
+    "Send an abort message for an incoming shell request"
+    # See https://github.com/ipython/ipykernel/blob/1d97cb2a04149387a0d2dbea1b3d0af691d8df6c/ipykernel/kernelbase.py#L623
+
+    parts <- zmq.recv.multipart(sockets$shell, unserialize = FALSE)
+    msg <- wire_to_msg(parts)
+    reply_type <- paste(strsplit(msg$header$msg_type, "_")[1], "_reply")
+    reply_content <- list(status = 'aborted')
+    send_response(reply_type, msg, "shell", reply_content)
+},
+
+abort_queued_messages = function(){
+    "Abort all already queued shell messages after an error"
+
+    while (TRUE) {
+        ret = zmq.poll(
+            c(sockets$shell), # only shell channel
+            c(.pbd_env$ZMQ.PO$POLLIN), # type
+            0) # no timeout, only what's there
+        if (ret != 1){
```

Comment:
According to the docs, zmq.poll should return two values: 'a ZMQ code and an errno'. So I suspect comparing it to a single integer is not the right thing to do. I'd be inclined to do it as an `else` clause of the bit below, that way there's only one thing we're checking instead of two.